src/foremast/utils/dns.py

```diff
@@ -104,14 +105,14 @@ def delete_existing_cname(env, zone_id, dns_name):
     startrecord = None
     newrecord_name = dns_name
     client = boto3.Session(profile_name=env).client('route53')
-    response = client.list_resource_record_sets(
-        HostedZoneId=zone_id
-    )
-    for item in response['ResourceRecordSets']:
-        if item['Name'].rstrip('.') == newrecord_name and item['Type'] == 'CNAME':
-            startrecord = item
-            LOG.info("Found old record: %s", item)
-            break
+    pager = client.get_paginator('list_resource_record_sets')
+
+    for rset in pager.paginate(HostedZoneId=zone_id):
+        for item in rset['ResourceRecordSets']:
```

Comment:
I would probably change variable from `item` to `record` as in the `primary_record_exists()` method. I think it makes it clearer.